content/guides/10-minute-tutorial.md

```diff
@@ -792,12 +793,281 @@ Feature: Is it Friday yet?
+
+{{% block "javascript" %}}
+```javascript
+function isItFriday(today) {
+  return (today === "Friday") ? "TGIF" : "Nope";
+}
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```ruby
+def is_it_friday?(day)
+  answer = (day == 'Friday') ? 'TGIF' : 'Nope'
+end
+```
+{{% /block %}}
+
+Run Cucumber again:
+
+{{% block "javascript" %}}
+```shell
+...
+1 scenario (1 passed)
+3 steps (3 passed)
+0m00.001s
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```shell
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday        # features/is_it_friday_yet.feature:4
+    Given today is Sunday              # features/step_definitions/stepdefs.rb:8
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:12
+    Then I should be told "Nope"       # features/step_definitions/stepdefs.rb:16
+
+1 scenario (1 passed)
+3 steps (3 passed)
+0m0.049s
+```
+{{% /block %}}
+
+# Multiple scenarios
+We've seen what happens when it *isn't* Friday, but what about when it *is* Friday? Fortunately, we've already written the statement to evaluate whether or not `today` is equal to `"Friday"` and return `"TGIF"` if it is. So, let's write a scenario where today is Friday!
+
+Update the `is-it-friday-yet.feature` file.
+```gherkin
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday
+    Given today is Sunday
+    When I ask whether it's Friday yet
+    Then I should be told "Nope"
+
+  Scenario: Friday is Friday
+    Given today is Friday
+    When I ask whether it's Friday yet
+    Then I should be told "TGIF"
+```
+
+{{% block "javascript" %}}
+Update the `stepdefs.js` file.
+```javascript
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+
+function isItFriday(today) {
+  return (today === "Friday") ? "TGIF" : "Nope";
+}
+
+// These two Given functions are very similar but, as written, require separate functions
+Given('today is Sunday', function () {
+  this.today = 'Sunday';
+});
+Given('today is Friday', function () {
+  this.today = 'Friday';
+});
+
+// This is a shared function and should be moved to a shared.js file
+When('I ask whether it\'s Friday yet', function () {
+  this.actualAnswer = isItFriday(this.today);
+});
+
+// This is also a shared function that uses a variable, {string}
+// This should also be moved to a shared.js file
+Then('I should be told {string}', function (expectedAnswer) {
+  assert.equal(this.actualAnswer, expectedAnswer);
+});
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```ruby
+module FridayStepHelper
+  def is_it_friday?(day)
+    answer = (day == 'Friday') ? 'TGIF' : 'Nope'
+  end
+end
+World FridayStepHelper
+
+Given("today is Sunday") do
+  @today = 'Sunday'
+end
+
+Given("today is Friday") do
+  @today = 'Friday'
+end
+
+# Then is a shared function
```

Comment:
What do these comments mean here? All step definitions are shared. Do you mean `When` here also?

I'd probably just remove these 2 comments

Reply:
So, I thought that the preferred way to do this was to have something like a shared_steps.rb for things like this. That is what that comment is getting at, but I'm happy to just remove these two comments since the next tutorial step is collapsing those two scenarios into a scenario outline with examples. I just wanted to briefly point out that those scenario steps actually share a single definition (since I did not think this was intuitive to someone who is brand new to cucumber).

```gherkin
Scenario: Common scenario 1
  Given an uncommon situation
  When a common thing happens
  Then do something common

Scenario: Common scenario 2
  # This requires a unique function:
  Given another uncommon situation
  # This function should already have a step def because it the same as from the first scenario, this is what I mean by a "shared" function
  When a common thing happens
  # Again, "shared" function
  Then do something common
```

Reply:
Let's remove the comments as I think it would raise more questions than clarify. If you want to clarify that steps can be reused, I'd prefer to do that in the text part of the tutorial (and preferably language agnostic).